src/cpp/fastdds/topic/DDSSQLFilter/DDSFilterField.hpp

```diff
@@ -89,30 +93,184 @@ struct DDSFilterField final : public DDSFilterValue
      * Performs the deserialization of the field represented by this DDSFilterField.
      * Will notify the predicates where this DDSFilterField is being used.
      *
-     * @param[in]  payload  The payload from where to deserialize this field.
+     * @param[in]  data  The deserialization of the payload being filtered.
      *
      * @return Whether the deserialization process succeeded
      *
      * @post Method @c has_value returns true.
      */
     bool set_value(
-            const eprosima::fastrtps::rtps::SerializedPayload_t& payload)
+            eprosima::fastrtps::types::DynamicData& data_value)
     {
-        // TODO: Set value from payload
-        static_cast<void>(payload);
+        return set_value(data_value, 0);
+    }
+
+    bool set_value(
+            eprosima::fastrtps::types::DynamicData& data,
+            size_t n)
+    {
+        using namespace eprosima::fastrtps::types;
 
-        has_value_ = nullptr != type_object_;
+        uint32_t index = static_cast<uint32_t>(access_path_[n].member_index);
+        auto member_id = data.get_member_id_at_index(index);
+        bool last_step = access_path_.size() - 1 == n;
+        bool ret = false;
 
-        // TODO: Inform parent predicates
+        if (access_path_[n].array_index < std::numeric_limits<MemberId>::max())
```

Comment:
Consider using the macro `MEMBER_ID_INVALID` defined in `fastrtps/types/TypesBase.h`.

Reply:
Commit 6727790